docs/choreolib/auto-routines.md

```diff
@@ -1,7 +1,92 @@
 # Auto Routines
 
+## Setup
+
+### AutoFactory
+
 ChoreoLib provides the `AutoFactory` class as higher level API to make it easier to create competitive and complex auto routines inside your robot code.
 
+You can setup the `AutoFactory` by calling `Choreo.createAutoFactory`
+
+```java
+// The most basic usage of the AutoFactory
+class Robot extends TimedRobot {
+  /** A swerve drive subsystem */
+  private final Drive drive = ...;
+  /** An object that manages information about robot position */
+  private final Localizer localizer = ...;
+  private final AutoFactory autoFactory;
+
+  public Robot() {
+    autoFactory = Choreo.createAutoFactory(
+      drive,
+      localizer::pose,
+      (p, sample) -> drive.drive(new ChassisSpeeds(sample.vx, sample.vy, sample.omega)),
```

Comment:
This is not a good controller function to copy.

Reply:
Ill add a comment that its bad but including a good one is out of scope of that particular example, maybe ill just black box it for the example.

Reply:
```suggestion
      drive::choreoController
```